pyomo/contrib/incidence_analysis/tests/test_interface.py

```diff
@@ -1791,6 +1791,64 @@ def test_linear_only(self):
         self.assertIs(matching[m.eq2], m.x[2])
         self.assertIs(matching[m.eq3], m.x[3])
 
+    def test_add_edge(self):
+        m = pyo.ConcreteModel()
+        m.x = pyo.Var([1, 2, 3, 4])
+        m.eq1 = pyo.Constraint(expr=m.x[1] ** 2 + m.x[2] ** 2 + m.x[3] ** 2 == 1)
+        m.eq2 = pyo.Constraint(expr=m.x[2] + pyo.sqrt(m.x[1]) + pyo.exp(m.x[3]) == 1)
+        m.eq3 = pyo.Constraint(expr=m.x[3] + m.x[2] + m.x[4] == 1)
+        m.eq4 = pyo.Constraint(expr=m.x[1] + m.x[2] ** 2 == 5)
+
+        igraph = IncidenceGraphInterface(m, linear_only=False)
+        n_edges_original = igraph.n_edges
+
+        # Test edge is added between previously unconnectes nodes
+        igraph.add_edge(m.x[1], m.eq3)
+        n_edges_new = igraph.n_edges
+        assert ComponentSet(igraph.get_adjacent_to(m.eq3)) == ComponentSet(m.x[:])
+        self.assertEqual(n_edges_original + 1, n_edges_new)
+
+        # Test no edge is added if there exists a previous edge between nodes
+        igraph.add_edge(m.x[2], m.eq3)
+        n_edges2 = igraph.n_edges
+        self.assertEqual(n_edges_new, n_edges2)
+
+    def test_add_edge_linear_igraph(self):
+        m = pyo.ConcreteModel()
+        m.x = pyo.Var([1, 2, 3, 4])
+        m.eq1 = pyo.Constraint(expr=m.x[1] + m.x[3] == 1)
+        m.eq2 = pyo.Constraint(expr=m.x[2] + pyo.sqrt(m.x[1]) + pyo.exp(m.x[3]) == 1)
+        m.eq3 = pyo.Constraint(expr=m.x[4] ** 2 + m.x[1] ** 3 + m.x[2] ** 2 == 1)
+
+        # Make sure error is raised when a variable is not in the igraph
+        igraph = IncidenceGraphInterface(m, linear_only=True)
+
+        msg = "is not a variable in the incidence graph"
+        with self.assertRaisesRegex(RuntimeError, msg):
+            igraph.add_edge(m.x[4], m.eq2)
+
+    def test_var_elim(self):
+        m = pyo.ConcreteModel()
+        m.x = pyo.Var([1, 2, 3, 4])
+        m.eq1 = pyo.Constraint(expr=m.x[1] ** 2 + m.x[2] ** 2 + m.x[3] ** 2 == 1)
+        m.eq2 = pyo.Constraint(expr=pyo.sqrt(m.x[1]) + pyo.exp(m.x[3]) == 1)
+        m.eq3 = pyo.Constraint(expr=m.x[3] + m.x[2] + m.x[4] == 1)
+        m.eq4 = pyo.Constraint(expr=m.x[1] == 5 * m.x[2])
+
+        igraph = IncidenceGraphInterface(m)
+        # Eliminate x[1] usinf eq4
```

Comment:
Typo: "usinf"

Reply:
Reported in the February crate-ci/typos update PR.